client/me/billing-history/transactions-table.jsx

```diff
@@ -105,19 +105,20 @@ class TransactionsTable extends React.Component {
 	}
 
 	serviceName = transaction => {
-		var item, name;
-
 		if ( ! transaction.items ) {
-			name = this.serviceNameDescription( transaction );
-		} else if ( transaction.items.length === 1 ) {
-			item = transaction.items[ 0 ];
-			item.plan = capitalPDangit( titleCase( item.variation ) );
-			name = this.serviceNameDescription( item );
-		} else {
-			name = <strong>{ this.props.translate( 'Multiple items' ) }</strong>;
+			return this.serviceNameDescription( transaction );
 		}
 
-		return name;
+		const groupedTransactionItems = groupDomainProducts( transaction.items, this.props.translate );
+
+		if ( groupedTransactionItems.length > 1 ) {
+			return <strong>{ this.props.translate( 'Multiple items' ) }</strong>;
+		}
+
+		return this.serviceNameDescription( {
+			...groupedTransactionItems[ 0 ],
```

Comment:
Let's avoid the magic number here,  either a variable name would be helpful here, or we can use an array destructure like so:
```jsx
const [ transactionItem, ...groupedItems ] = groupDomainProducts( transaction.items, this.props.translate );
if ( groupedItems.length > 0 ) {
    // multiple items
}

return this.serviceNameDescription( {
    ...transactionItem,
    plan: capitalPDangit( titleCase( transactionItem.variation ) ),
} );
```

Reply:
It's not a magic number, but the actual first element of a 1-element array. 🙂 

The idea is that, if the (grouped) receipt contains only 1 item (of any type), we show that item's details in the receipts list.

On the other hand, if the receipt contains multiple items (again, of any type), we show "Multiple items".
In this case, we could have, for example:
- 1 "normal" item (e.g. "WP.com Biz subscription") + 1 domain item
- 2 normal items + 0 domain items
- 0 normal items + 2 domain items

So, in your example, `[ transactionItem, ...groupedItems ]` could end up with a semantically wrong destructuring.
For example, if there are 2 `transactionItems`, the second would end up into the `groupedItems` array, which should only contain domain groups.

The result would be the same, but in my opinion using the first and only element of a length-1 array should be clear enough. 🙂

Reply:
>  first element of a 1-element array

For readability a variable name is still very helpful! When folks are passing back arrays of misc values not having to dig through the generating function is great.